service/src/main/kotlin/fi/espoo/evaka/vasu/VasuController.kt

```diff
@@ -135,7 +161,51 @@ class VasuController(
             dbc.transaction { tx ->
                 val vasu = tx.getVasuDocumentMaster(id) ?: throw NotFound("vasu $id not found")
                 validateVasuDocumentUpdate(vasu, body)
-                tx.updateVasuDocumentMaster(id, body.content, body.childLanguage)
+
+                val transformedFollowupsVasu = body.copy(
+                    content = body.content.copy(
+                        sections = body.content.sections.mapIndexed { sectionIndex, section ->
+                            section.copy(
+                                questions = section.questions.mapIndexed { questionIndex, question ->
+                                    if (question.type === VasuQuestionType.FOLLOWUP) {
+                                        val followup = question as VasuQuestion.Followup
+                                        val storedFollowup = vasu.content.sections[sectionIndex].questions[questionIndex] as VasuQuestion.Followup
+
+                                        followup.copy(
+                                            value = followup.value.mapIndexed { entryIndex, entry ->
+                                                val storedEntry = storedFollowup.value.getOrNull(entryIndex)
+                                                entry.copy(
+                                                    authorId = storedEntry?.authorId ?: EmployeeId(user.rawId()),
+                                                    edited =
+                                                    if (
+                                                        storedEntry == null || (
+                                                            storedEntry.authorId == EmployeeId(user.rawId()) && storedEntry.createdDate?.let {
+                                                                HelsinkiDateTime.now().durationSince(it).toMinutes() < 15
+                                                            } == true
+                                                            ) || (entry.text == storedEntry.text && entry.date.isEqual(storedEntry.date))
+                                                    )
+                                                        storedEntry?.edited
+                                                    else
+                                                        FollowupEntryEditDetails(
+                                                            editedAt = LocalDate.now(),
+                                                            editorId = EmployeeId(user.rawId()),
+                                                            editorName = null
+                                                        ),
```

Comment:
Same here: I think it wouldn't hurt to use named functions or variables here to add more semantics for better readability